### DISCUSSION TITLE
Make prop for prepend / append icon color

### DISCUSSION THREAD

**Discussion by Joebayld** [vuetifyjs/vuetify#19101] [Ideas]
I find that I'm constantly styling my buttons to have just the icon colored. I'm using the following code to do so:

```vue
<template v-slot:prepend>
  <v-icon color="success"></v-icon>
</template>
```

While this works fine, it ends up being a lot of code in the project just to change colors of the icon. Is it possible to add a prop to the buttons to color this? Even if I extend the component myself and it's only for my project? Ideally it would be in the API but that's not my call. 

Looking for something like: 

```vue
 <v-btn  
  prepend-icon="mdi-close-circle" 
  prepend-icon-color="red">
    Close
</v-btn>
```

**Comment by yuwu9145** [comment]
Are you trying to set icon colour different from default text colour?

**Reply by Joebayld** [reply]
Yes exactly. Keeping the text the default and just color the icon. 

<img width="516" alt="Screenshot 2024-01-24 at 4 33 16 PM" src="https://github.com/vuetifyjs/vuetify/assets/9399317/3741af6d-2f05-4ddd-a12f-0be2c09810b0">

**Reply by yuwu9145** [reply]
Try `defaults-provider`. [demo](https://play.vuetifyjs.com/#eNplULtOxDAQ/JXFjUHCiU5wTZRDQjTwA1AQimBvkCW/tHbSRPl3bIfjIbqd8c54Zl9XFkm29yE0y4ysY31CG8yY8G5wAP0ixhDqWIH0Lo3aIX1RlVQ4jbNJUQTyi1ZI0J2p08BW/vyUZbxbufTGE+94nKXEGPm2DezbqFq9JweBMKBTQmdV1lulhTQ+opCapMEsgYeCoW+r4CdKxv+ynLPnt7/hC7N369tfnTOMknRIEDHNe3Vtg6cEKxBOsMFE3gLP5+JlHSD7xgQ2fsCpbFzyRzTGw4snoy74Vflgt8xmbLtmN81tcziyMhybA3v7BLAgfu8=)

![image](https://github.com/vuetifyjs/vuetify/assets/5698884/e1b29962-25a1-4713-9df5-bcd515994c48)

**Reply by Joebayld** [reply]
Thanks for this. 

Is there a way to inject a prop on each component though? I still would like the ability to modify the colors per button but with less code. 

Could this help added to the standard API?

**Reply by yuwu9145** [reply]
`defaults-provider` is designed to inject default prop values to each child component, it should cover most cases.

Was there any use case in your project that cannot be resolved by it?

**Reply by KaelWD** [reply]
https://vuetifyjs.com/en/features/global-configuration/#contextual-defaults
